BUILD.md

```diff
@@ -0,0 +1,472 @@
+components you are interested in building, these instructions assume you have
+the following tools already setup:
+- [Go](https://golang.org)
+- [Rust](https://www.rust-lang.org)
+- [Minikube](https://github.com/kubernetes/minikube)
+- [Docker](https://www.docker.com/)
+- [Yarn](https://yarnpkg.com)
+
+# Repo layout
+
+Conduit is primarily writtern in Go and Rust (and JavaScript!). At its core
+is a high-performance data plane written in Rust. The control plane components
+are writtern in Go. The dashboard UI is a React application.
+
+## Control Plane (Go)
+
+- [`cli`](cli): Command-line `conduit` utility, view and drive the control
+  plane.
+- [`controller`](controller)
+  - [`destination`](controller/destination): Serves service discovery
+    information to the `proxy`.
+  - [`proxy-api`](controller/api/proxy): Accepts requests from `proxy`
+    instances and forwards those requests to the appropriate controller
+    service.
+  - [`public-api`](controller/api/public): Accepts requests from API
+    clients such as `cli` and `web`, provides access to and control of the
+    conduit service mesh.
+  - [`tap`](controller/tap): Provides a live pipeline of requests.
+  - [`telemetry`](controller/telemetry): Collects and aggregates
+    metrics from `proxy` componenets.
+- [`proxy-init`](proxy-init): Adds a Kubernetes pod to join the Conduit
+  Service Mesh.
+- [`web`](web): Provides a UI dashboard to view and drive the control plane.
+
+## Data Plane (Rust)
+
+  - [`proxy`](proxy): High-performance data plane, injected as a sidecar with
+    every service.
+  - [`tower-grpc`](tower-grpc): A client and server gRPC implementation based on
+    Tower.
+  - [`tower-grpc-examples`](tower-grpc-examples): Demonstrates how to use Tower
+    gRPC clients and servers with code generation.
+  - [`tower-h2`](tower-h2): Tower `Service` abstractions for HTTP/2 and Rust.
+  - [`tower-router`](tower-router): A Tower middleware that routes requests to
+    one of a set of inner services using a request predicate.
+
+# Components
+
+![Conduit Components](https://g.gravizo.com/source/svg/conduit_components?https%3A%2F%2Fraw.githubusercontent.com%2Frunconduit%2Fconduit%2Fsiggy%2Fbuild%2FBUILD.md)
+
+<details>
+<summary></summary>
+conduit_components
+  digraph G {
+    rankdir=LR;
+
+    node [style=filled, shape=rect];
+
+    "cli" [color=lightblue];
+    "destination" [color=lightblue];
+    "proxy-api" [color=lightblue];
+    "public-api" [color=lightblue];
+    "tap" [color=lightblue];
+    "telemetry" [color=lightblue];
+    "web" [color=lightblue];
+
+    "proxy" [color=orange];
+
+    "cli" -> "public-api";
+    "web" -> "public-api";
+
+    "destination" -> "kubernetes";
+
+    "proxy" -> "proxy-api";
+
+    "proxy-api" -> "destination";
+    "proxy-api" -> "telemetry";
+
+    "public-api" -> "tap";
+    "public-api" -> "telemetry";
+
+    "tap" -> "proxy";
+
+    "telemetry" -> "prometheus";
```

Comment:
Technically every go service has a dependency on prometheus, since prometheus scrapes metrics from each service and those metrics are reported on the Service Mesh overview page in the web UI. But as far as the data plane => control plane interaction goes, only telemetry writes to prometheus, which is what you have here. I'm a bit torn on whether or not it would be worth reflecting the other dependencies. It would probably complicate the graph, so fine as is, but just wanted to mention it.